android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactory.java

```diff
@@ -79,33 +89,79 @@
         final MaterialEditText referenceEditText = rootLayout.findViewById(R.id.reference_edit_text);
         final String referenceEditTextHint = jsonObject.optString(REFERENCE_EDIT_TEXT_HINT, context.getString(R.string.enter_number_of_repeating_group_items));
         final String repeatingGroupLabel = jsonObject.optString(REPEATING_GROUP_LABEL, context.getString(R.string.repeating_group_item));
+        String remoteReferenceEditText = jsonObject.optString(REFERENCE_EDIT_TEXT);
+
+        // Enables us to fetch this value from a previous edit_text & disable this one
+        retrieveRepeatingGroupCountFromRemoteReferenceEditText((JsonApi) context, referenceEditText, remoteReferenceEditText);
         setUpReferenceEditText(referenceEditText, referenceEditTextHint, repeatingGroupLabel);
 
-        doneButton = rootLayout.findViewById(R.id.btn_repeating_group_done);
-        doneButton.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                addOnDoneAction(referenceEditText);
-            }
-        });
+        // Disable the done button if the reference edit text being used is remote & has a valid value
+        if (isRemoteReferenceValueUsed) {
+            doneButton.setVisibility(View.GONE);
+        } else {
+            doneButton.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    addOnDoneAction(referenceEditText);
+                }
+            });
+        }
 
         ((JsonApi) context).addFormDataView(referenceEditText);
 
         return views;
     }
 
+    private void retrieveRepeatingGroupCountFromRemoteReferenceEditText(@NonNull JsonApi context, @NonNull MaterialEditText referenceEditText, @Nullable String remoteReferenceEditTextAddress) throws JSONException {
+        if (!TextUtils.isEmpty(remoteReferenceEditTextAddress) && remoteReferenceEditTextAddress.contains(":")) {
+            String finalRemoteReferenceEditTextAddress = remoteReferenceEditTextAddress.trim();
+            String[] addressSections = finalRemoteReferenceEditTextAddress.split(":");
+            String remoteReferenceEditTextStep = addressSections[0];
+            String remoteReferenceEditTextKey = addressSections[1];
+            JSONObject stepJsonObject = context.getmJSONObject().optJSONObject(remoteReferenceEditTextStep);
+
+            if (stepJsonObject != null) {
+                JSONArray fields = stepJsonObject.optJSONArray(FIELDS);
+
+                for (int i = 0; i < fields.length(); i++) {
+                    JSONObject stepField = fields.optJSONObject(i);
+                    if (stepField != null && stepField.has(KEY) && stepField.getString(KEY).equals(remoteReferenceEditTextKey)) {
+                        String fieldValue = stepField.optString(VALUE);
+
+                        if (!TextUtils.isEmpty(fieldValue)) {
```

Comment:
Use `StringUtils.isNotBlank` it even checks for white spaces https://javakeypoint.wordpress.com/2018/01/18/stringutils-isnotempty-vs-stringutils-isnotblank/